### PR TITLE
feat: accessibility addition suggestions for pagination

### PR DESCRIPTION
**Pull request by Martskin**
Accessibility addition suggestions for pagination

#### Description

Adds aria attributes to pagination controls for a11y. 

- `aria-haspopup="true"`
- `aria-expanded="false"`
- `aria-label="next page. current page 2 of 20"`
- `role="menu"`

#### Scope

- [ ] Patch: Bug Fix
- [x] Minor: New Feature
- [ ] Major: Breaking Change

### CODE REVIEW

File: src/pagination/pagination.js [hunk 1]
```diff
@@ -175,33 +211,43 @@ export default class Pagination extends React.PureComponent<
...
-                      component: StyledDropdownMenu,
-                      // Access $style manually because it has gone through transformation
-                      // from the override helper function already
-                      // $FlowFixMe
-                      style: dropdownMenuProps.$style,
-                    },
-                  }}
-                  {...dropdownMenuProps}
-                />
+                <div role="menu">
```
Comment (chasestarr) [inline src/pagination/pagination.js]: The [dropdown menu](https://github.com/uber-web/baseui/blob/master/src/menu/menu.js#L31) should have a `listbox` role. Is `menu` more appropriate? If so, we try to allow consumers of baseui to customize all 'sub-components'. can the role be passed to `List` override?

```jsx
<DropdownMenu
  overrides={{
    List: {
      props: {role: 'menu'},
    },
  }}
/>
```

Comment (Martskin) [reply src/pagination/pagination.js]: Thanks @chasestarr !
Ahh, I see. Good suggestion.. I changed the role to an override and I added an override for the options as well to be `menuitem`.

File: src/pagination/pagination.js [hunk 2]
```diff
@@ -97,6 +101,23 @@ export default class Pagination extends React.PureComponent<
     this.setState({isMenuOpen});
   };
 
+  constructAriaWayfinderLabel = (locale: object, prefix: string) => {
```
Comment (Martskin) [inline src/pagination/pagination.js]: @chasestarr 
I know I'm not doing this type checking correctly. Any suggestions?

Comment (chasestarr) [reply src/pagination/pagination.js]: `object` can be replaced with `LocaleT` from [here](https://github.com/uber-web/baseui/blob/26a8b9a38ef5621d5055fd185ca3a88b5a8804ca/src/locale/types.js)

**Comment by Martskin** [comment]
My pleasure! Thanks for merging! @chasestarr

**Pull request merged**